### PR TITLE
Fix negative venation estimation at lock

### PR DESCRIPTION
**Pull request by johnmark13**
Again, maybe not exactly where we want to go, but this will prevent the manual entry of dates in earlier than the lock date meaning that now the slider, date control and input text are guarded.

Dework -https://app.dework.xyz/nation3/citizen-app-p?taskId=f614e270-4baf-4e1a-80d7-a6a9f3ad1234

### CODE REVIEW

**Comment by aahna-ashina** [comment]
Adding screenshots for context:

### Before

https://app.nation3.org/lock

Screen Shot 2022-08-11 at 7 38 45 PM

### After

https://app-a59xtx3yy-nation3.vercel.app/lock

Screen Shot 2022-08-11 at 7 38 57 PM

File: ui/pages/lock.tsx [hunk 1]
```diff
@@ -133,9 +133,7 @@ export default function Lock() {
   useEffect(() => {
     if (hasLock && veNationLock) {
       setMinMaxLockTime({
-        min: dateToReadable(
-          oneWeekOut
-        ),
+        min: dateToReadable(bigNumberToDate(veNationLock[1])),
```
Comment (aahna-ashina) [inline ui/pages/lock.tsx:136]: What is the difference between `veNationLock[0]` and `veNationLock[1]`?

File: ui/pages/lock.tsx [hunk 2]
```diff
@@ -365,6 +363,7 @@ export default function Lock() {
                     min={minMaxLockTime.min}
                     max={minMaxLockTime.max}
                     onChange={(e: any) => {
+                      if(e.target.value < minMaxLockTime.min) { return false;}
```
Comment (aahna-ashina) [inline ui/pages/lock.tsx:366]:
```suggestion
                      if (e.target.value < minMaxLockTime.min) {
                        return false;
                      }
```

File: ui/pages/lock.tsx (hunk 1 above)
Comment (johnmark13) [reply ui/pages/lock.tsx:136]: It is weakly typed tuple, the first value `[0]` holds the amount the other `[1]` the date of the existing lock.

**Review by aahna-ashina — CHANGES_REQUESTED** [review]
This seems like an improvement to me, in terms of UX/UI.  And if we make this adjustment, we also need to update the explainer text from "Minimum one week, maximum four years from now." to something else, because the "Minimum one week" does no longer apply.

> Screen Shot 2022-08-11 at 7 54 13 PM

File: ui/pages/lock.tsx (hunk 2 above)
Comment (johnmark13) [reply ui/pages/lock.tsx:366]: I much prefer this (your) syntax, but we should probably have prettier or something to enforce it? (When i run `npm run lint` the line you changed is not flagged as problematic.

File: ui/pages/lock.tsx (hunk 1 above)
Comment (aahna-ashina) [reply ui/pages/lock.tsx:136]: @johnmark13 Ok, then maybe we can introduce a variable like `endDate = veNationLock[1]` or something like that, for increased code readability?

**Comment by aahna-ashina** [comment]
@johnmark13 Remember to include a link to the Dework task in your pull request descriptions.  That will save the reviewers some time from having to manually search up the task in Dework.

File: ui/pages/lock.tsx (hunk 1 above)
Comment (johnmark13) [reply ui/pages/lock.tsx:136]: I am happy too, but remember that this proposal was only a quick get out of jail before we refactor the whole tsx file and veNationLock[0|1] is all over the file.

Comment (johnmark13) [reply ui/pages/lock.tsx]: Made the change, local to the call, but can move to be a top level constant if you prefer?

**Review by xPi2 — APPROVED** [review]
Looks good as first quick approach

**Review by aahna-ashina — CHANGES_REQUESTED** [review]
Screen Shot 2022-08-16 at 12 44 20 PM

> The minimum lock expiration date is the greater of one week from now or, if you have an existing lock, your existing lock expiration date. The maximum lock time is four years from now.

@johnmark13 This is a very long text, which will cause more confusion.  Is it possible to make _two variants_ of this text, one for users that have never locked before, and another for users who already have an initial lock?

Can this be done with an `if-else`?
```
if (noExistingLock) {
  "Minimum one week, maximum four years from now."
} else {
  "Maximum four years from your existing lock expiration date."
}
```

**Review by aahna-ashina — APPROVED** [review]

**Pull request merged**